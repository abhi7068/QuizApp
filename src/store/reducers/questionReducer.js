import * as actionTypes from "../actions/actionTypes";
const initialState = {
    quiz: [],
};
const questionReducer = (state = initialState, action) => {
    console.log(action);
    switch (action.type) {
        case actionTypes.ADD_QUESTION:
            console.log(action)
            return {
                ...state,
              quiz:[...state.quiz,action.question]
            };
        default:
            return state;
    }
};

export default questionReducer;