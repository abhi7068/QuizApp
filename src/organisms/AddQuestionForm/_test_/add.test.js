import React from 'react'
import AddQuestionsForm from "../index";
import { render, cleanup,fireEvent } from  '@testing-library/react';
import '@testing-library/jest-dom';
import { createStore} from 'redux';
import { Provider } from 'react-redux';
import  questionReducer from "../../../store/reducers/questionReducer" 
import { BrowserRouter as Router, withRouter } from 'react-router-dom';
const store=createStore(questionReducer);
afterEach(cleanup);

 
  it('should take a snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <AddQuestionsForm  />
      </Provider>
    );
    expect(
      asFragment(
        render(
          <Provider store={store}>
            <AddQuestionsForm  />
          </Provider>
        )
      )
    ).toMatchSnapshot();
  });

it('it renders', () => {
  render(
   <Provider store={store}>
          <Router>
        <AddQuestionsForm />
        </Router>
      </Provider>
  );
 });
 
describe("Testing handler", () => {
    test("test", () => {
        const { getByTestId } = render(
            <Provider store={store}>
                <Router>
                    <AddQuestionsForm />
                </Router>
            </Provider>
        );
        const name = getByTestId("Input");
      // const answer = getByTestId("output");
       // const  = getByTestId("name");
       fireEvent.change(name, { target: { value: "welcome" } });
      // fireEvent.change(answer, { target: { value: "welcome1" } });
        //fireEvent.change(name,{ target: { value: "welcome" } });
        expect(getByTestId("Input").value).toEqual("welcome");
      // expect(answer.value).toEqual("welcome1");
    
    });
     
})


describe("Testing handler2", () => {
  test("test", () => {
      const { getByTestId } = render(
          <Provider store={store}>
              <Router>
                  <AddQuestionsForm />
              </Router>
          </Provider>
      );
      const answer = getByTestId("output");
    // const answer = getByTestId("output");
     // const  = getByTestId("name");
     fireEvent.change(answer, { target: { value: "welcome" } });
    // fireEvent.change(answer, { target: { value: "welcome1" } });
      //fireEvent.change(name,{ target: { value: "welcome" } });
      expect(getByTestId("output").value).toEqual("welcome");
    // expect(answer.value).toEqual("welcome1");
  
  });
   
})