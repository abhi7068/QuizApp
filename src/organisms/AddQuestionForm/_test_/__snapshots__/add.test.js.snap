// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <form>
      <div
        class="MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
            data-shrink="false"
            for="content"
            id="content-label"
          >
            Add Question
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
              â€‰*
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              data-testid="Input"
              id="content"
              name="question"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <br />
      <br />
      <div
        class="MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Add Option
          </span>
        </button>
      </div>
      <br />
      <br />
      <div
        class="MuiGrid-root"
      />
      <br />
      <br />
      <div
        class="MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Submit
          </span>
        </button>
        <br />
        <br />
      </div>
    </form>
  </div>
</DocumentFragment>
`;
